### ISSUE TITLE
WinLog INPUT: include the StringInserts key-value pairs into the log record

### ISSUE THREAD

**Issue by jsubirat** [fluent/fluent-bit#2427]
**Is your feature request related to a problem? Please describe.**
In [this pull request](https://github.com/fluent/fluent-bit/pull/2322),  the `StringInserts` [were removed from the resulting log record](https://github.com/fluent/fluent-bit/pull/2322/files#diff-0890d3b5666d8c56708c223ac7bc54a3L267-L269). A formatted `Message` containing the human-readable message [was included instead](https://github.com/fluent/fluent-bit/pull/2322/files#diff-44387dae255041c828cb88280efd027fR401-R406).

This solution is really useful to visualize the resulting message, but it would also be good to include the key-value pairs present in the `StringInserts` as log record attributes.

In FluentBit 1.4.1, a `winlog` contained a `StringInserts` field like the following:
```
[42] winlog.0: [1596733503.081829800, {"RecordNumber"=>43, "TimeGenerated"=>1585850266, "TimeWritten"=>1585850266, "EventID"=>600, ...
"StringInserts"=>["Variable", "Started", "        ProviderName=Variable
        NewProviderState=Started

        SequenceNumber=11

        HostName=ConsoleHost
        HostVersion=5.1.18362.145
...
        RunspaceId=
        PipelineId=
        CommandName=
        CommandType=
        ScriptName=
        CommandPath=
        CommandLine="], "Sid"=>"", "Data"=>""}]
```

It would be really nice if the resulting log record contained all the key-value pairs present in the `StringInserts`, that is, `NewProviderState`, `SequenceNumber`,  `HostName`, `HostVersion`, `RunspaceId` **(even if it is empty)**...

**Describe the solution you'd like**
Include all the key-value pairs present in `StringInserts` as log record attributes. 

**Additional context**
Even though the `Message` field is very useful as it is human-readable, including all the `StringInserts` key-value pairs as log attributes would enable the user to filter by these registry key values more easily later.

**Comment by fujimotos** [comment]
Hmm. I can bring back StringInserts to the output messages
if there are demands for it, but:

> It would be really nice if the resulting log record contained all the key-value pairs present in the StringInserts, that is, NewProviderState, SequenceNumber, HostName, HostVersion, RunspaceId (even if it is empty)...

I think it is hard to parse the field the way as you described.

The reason is that `StringInserts` is just an array of arbitrary strings,
which do not have a consistent format. In fact, most log messages do
not even contain such key/value pairs e.g.

```
winlog.0: [1597210850.154986300, {..."StringInserts"=>["0", "2020-08-11T10:58:30.760889700Z"], ...}
```

Possibly you can apply `filter_parser` to parse the key/value pairs
from event messages from a certain source, but I don't believe it
can be implemented in a general way in `in_winlog`.

**Comment by edsiper** [comment]
@fujimotos I think we should find a way to solve the data processing, I think this needs 3 steps:

__1__. revert back this code:

https://github.com/fluent/fluent-bit/pull/2322/files#diff-0890d3b5666d8c56708c223ac7bc54a3L267-L279

__2__ add new boolean option called ```stringinserts```, enabled by default. If set, include the values as part of the array (like it used to work), if disabled, just skip that field.

__3__ offer a built-in parser: here I would like to hear more from the community about use cases so we can come up with the proper solution.

**Comment by fazal-10** [comment]
As we know,in StringInserts ,data is received in array format.Is there any way to receive the data in json format instead of array?

is there any doc or website which has any information related to it, if anyone has, plz share it here.

one solution is writing script to convert array to json,but if there are any effective solution exist,then i would like to chose that solution.

so,if anyone has any idea about my problem, plz share your thoughts,it will be really helpful for me.

Thanks in advance!
